File: LeetCode/Tag/Two-Pointers/cpp/11.container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& A) {
        int l = 0, r = A.size()-1, maxArea = 0;
        while(l < r) {
            maxArea = max((r-l) * min(A[l], A[r]), maxArea);
            if(A[l] < A[r])
                ++l;
            else
                --r;
        }
        return maxArea;
    }
};

File: LeetCode/Tag/Two-Pointers/js/11.container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0, right = height.length - 1;
    let max = 0;
    while(left < right){
        let w = right - left;
        let h = Math.min(height[left],height[right]);
        max = Math.max(max, w * h);

        if(height[left] >= height[right]){
            right--;
        }
        else{
            left++;
        }
    }
    return max;
};
